### PR TITLE
Use Ranch for TCP acceptor pool

### PR DESCRIPTION
**Pull request by essen**
Part of https://github.com/rabbitmq/rabbitmq-server/issues/260.

I didn't add a dependency on Ranch yet; I suspect rabbit_common
is used by more than the server, and we would depend on Ranch
even for unrelated things. I will need to check this before
merging.

### CODE REVIEW

**Comment by michaelklishin** [comment]
`rabbitmq_common` is shared by the server and our Erlang client.

**Comment by michaelklishin** [comment]
Testing GH notifications.

**Comment by michaelklishin** [comment]
Ditto.

**Comment by michaelklishin** [comment]
@essen the change doesn't seem to add a dependency on Ranch. Is this intentional?

**Comment by michaelklishin** [comment]
OK, `rabbit` does depend on Ranch.

**Comment by essen** [comment]
Yeah I'm not sure we should depend on Ranch from common, considering it's also used by the client which does not (and should not have to) need Ranch.

File: src/rabbit_reader.erl [hunk 1]
```diff
@@ -340,13 +345,11 @@ start_connection(Parent, HelperSup, Deb, Sock, SockTransform) ->
            end,
     {ok, HandshakeTimeout} = application:get_env(rabbit, handshake_timeout),
     InitialFrameMax = application:get_env(rabbit, initial_frame_max, ?FRAME_MIN_SIZE),
-    ClientSock = socket_op(Sock, SockTransform),
-    erlang:send_after(HandshakeTimeout, self(), handshake_timeout),
```
Comment (michaelklishin) [inline src/rabbit_reader.erl]: @essen how are handshake timeouts handled now?

**Pull request merged**